lib/delivery-locations-resolver.js

```diff
@@ -0,0 +1,117 @@
+      }
+    ].map((loc) => Object.assign(loc, { label: `${loc.label}${customerCode !== 'PA' ? ' (mocked)' : ''}` }))
+  }
+}
+
+function recapCustomerCodesByBarcodes (barcodes) {
+  let scsbClient = new SCSBRestClient({ url: process.env.SCSB_URL, apiKey: process.env.SCSB_API_KEY })
+
+  // Record time to process all:
+  var __startAll = new Date()
+
+  return Promise.all(
+    barcodes.map((barcode) => {
+      // Record start time to process this request
+      var __start = new Date()
+      return scsbClient.searchByParam({ fieldValue: barcode, fieldName: 'Barcode' })
+        .then((response) => {
+          let ellapsed = ((new Date()) - __start)
+          logger.debug({ message: `HTC searchByParam API took ${ellapsed}ms`, metric: 'searchByParam-barcode', timeMs: ellapsed })
+
+          if (response && response.length > 0 && (typeof response[0]) === 'object') {
+            return { [barcode]: response[0].customerCode }
+          }
+        })
+        .catch((error) => {
+          // This is a common error:
+          //  "Error hitting SCSB API 502: <html>\r\n<head><title>502 Bad Gateway</title></head>\r\n<body bgcolor=\"white\">\r\n<center><h1>502 Bad Gateway</h1></center>\r\n</body>\r\n</html>\r\n"
+          // return Promise.reject(error)
+          logger.info({ message: 'HTC API error. Send everything to NH', htcError: error.message })
+          return { [barcode]: 'NH' }
+        })
+    })
+  ).then((scsbResponses) => {
+    let ellapsed = ((new Date()) - __startAll)
+    logger.debug({ message: `HTC searchByParam API across ${barcodes.length} barcodes took ${ellapsed}ms total`, metric: 'searchByParam-barcode-multiple', timeMs: ellapsed })
+
+    // Filter out anything `undefined` and make sure at least one is valid:
+    var validPairs = [{}].concat(scsbResponses.filter((r) => r))
+    // Merge array of hashes into one big lookup hash:
+    return Object.assign.apply(null, validPairs)
+  })
+}
+
+function resolveDeliveryLocations (items) {
+  // Extract barcodes from items:
+  var barcodes = items.map((i) => i.identifier.filter((i) => /^urn:barcode:/.test(i))[0].split(':')[2])
+
+  // Get a map from barcodes to ReCAP customercodes:
+  return recapCustomerCodesByBarcodes(barcodes)
+    .then((barcodeToCustomerCode) => {
+      // Now map over items to affix deliveryLocations:
+      return items.map((item) => {
+        // Get this item's barcode:
+        var barcode = item.identifier.filter((i) => /^urn:barcode:/.test(i))[0].split(':')[2]
+
+        // If recap has a customer code for this barcode, map it by recap cust code:
+        if (barcodeToCustomerCode[barcode]) {
+          item.deliveryLocation = deliveryLocationsByCustomerCode(barcodeToCustomerCode[barcode])
+
+        // Otherwise, it's not in recap:
+        } else {
+          // Temporarily use deliveryLocation if it's indexed (deprecated) - otherwise populate by mapping:
+          if (!item.deliveryLocation) {
```

Comment:
Yep. Now we're using the by-sierra-location mapping any time we have a holdingLocation to do the mapping. Still falling back on mocked data if we do not.